test/mocks/overrides/override_comment_view_model.dart

```diff
@@ -0,0 +1,51 @@
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/ui/comment_post.dart";
+import "package:proxima/viewmodels/comment_view_model.dart";
+
+import "../data/post_comment.dart";
+
+/// A mock implementation of the [CommentViewModel] class.
+/// By default it exposes an empty list of [CommentPost] and does nothing on refresh.
+class MockCommentViewModel
+    extends AutoDisposeFamilyAsyncNotifier<List<CommentPost>, PostIdFirestore>
+    implements CommentViewModel {
+  final Future<List<CommentPost>> Function(PostIdFirestore arg) _build;
+  final Future<void> Function() _onRefresh;
+
+  MockCommentViewModel({
+    Future<List<CommentPost>> Function(PostIdFirestore arg)? build,
+    Future<void> Function()? onRefresh,
+  })  : _build =
+            build ?? ((PostIdFirestore arg) async => List<CommentPost>.empty()),
+        _onRefresh = onRefresh ?? (() async {});
+
+  @override
+  Future<List<CommentPost>> build(PostIdFirestore arg) => _build(arg);
```

Comment:
I'm also not sure of the name `arg` here and line 13.

Reply:
I don't seem to get a warning here when renaming the parameter (I believe that is because this is a test file).
However I think that for consistency we might want to keep it named `arg` if we need to keep the parameter of `CommentViewModel.build` named `arg`.